File: lib-algorithm/src/main/java/com/notbad/lib/algorithm/LogUtils.java
package com.notbad.lib.algorithm;

public class LogUtils {
    private static final String TAG = "NotBad_";

    public static void d(String msg) {
        d("", msg);
    }

    public static void d(String tag, String msg) {
        System.out.println(TAG + tag + ":" + msg);
    }

    public static void i(String tag, String msg) {
        System.out.println(TAG + tag + ":" + msg);
    }

    public static void w(String tag, String msg) {
        System.out.println(TAG + tag + ":" + msg);
    }

    public static void e(String tag, String msg) {
        System.out.println(TAG + tag + ":" + msg);
    }

    public static void e(String tag, String msg, Exception e) {
        System.out.println(TAG + tag + ":" + msg + " error:" + e.getMessage());
    }
}

File: lib-common/src/main/java/com/notbad/lib/common/LogUtils.kt
package com.notbad.lib.common

import android.util.Log
import java.lang.Exception

object LogUtils {
    private const val PREF = "NotBad_"
    @JvmStatic
    fun d(tag:String,msg:String) {
        Log.d("$PREF$tag",msg)
    }
    @JvmStatic
    fun i(tag:String,msg:String) {
        Log.d("$PREF$tag",msg)
    }
    @JvmStatic
    fun w(tag:String,msg:String) {
        Log.w("$PREF$tag",msg)
    }
    @JvmStatic
    fun e(tag:String,msg:String,tr:Throwable?=null) {
        Log.e("$PREF$tag",msg,tr)
    }
}